### ISSUE TITLE
LDAP support

### ISSUE THREAD

**Issue by skddc** [sockethub/sockethub#249]
We're currently in the process of migrating all Kosmos services to unified LDAP accounts, so that it's easy to manage accounts across platforms and services.

Personally, I'd hate sending the user password to Sockethub for this, but perhaps there are better ways of proving to a Sockethub server that you own a certain LDAP account. Maybe a user could generate a token on their side and then upload it to their LDAP profile for example.

Basic support for search operations (per session I guess) is rather simple. There's for example https://www.npmjs.com/package/ldapjs, which we're going to use in https://gitea.kosmos.org/kosmos/akkounts-api.

**Comment by raucao** [comment]
> the original Kosmos context has moved on

What does that mean? It's still the same open issue. We do not have any auth yet, meaning the server is open to anyone.

**Comment by silverbucket** [comment]
@raucao Yeah, that doesn't make any sense, I have removed that triage message, and bumped this to medium priority.

**Comment by raucao** [comment]
ACK, thanks

FWIW, I'm still not sure what they best way forward is in regards to this problem. We should brainstorm and discuss it in person.

**Comment by silverbucket** [comment]
Triage: linked as the Phase 2 sub-issue of the auth roadmap (#735). It depends on Phase 1 (#1054), which delivers the token store and handshake/HTTP middleware this backend plugs into; the `POST /auth/token` exchange endpoint is introduced here.
